src/components/Expandable.tsx

```diff
@@ -18,16 +19,20 @@ const Expandable: React.FC<ExpandableProps> = ({
 }) => {
   const [expanded, setExpanded] = useState(false);
 
-  const toggleExpanded = useCallback(() => {
-    setExpanded(expanded => {
-      if (!expanded && activeIndex !== index && setActiveIndex)
+  const isExpanded =
+    (expanded && index === undefined) || (expanded && index === activeIndex);
+
+  const onToggleExpanded = useCallback(() => {
+    setExpanded(currentlyExpanded => {
+      if (!currentlyExpanded && activeIndex !== index && setActiveIndex)
         setActiveIndex(index);
-      return !expanded;
+      return !currentlyExpanded;
     });
   }, [setActiveIndex, activeIndex, index]);
 
-  const isExpanded =
-    (expanded && index === undefined) || (expanded && index === activeIndex);
+  useEffect(() => {
```

Comment:
@meissadia it's considered an anti-pattern to derive state from props, would it make more sense to use `useMemo` if you're concerned about performance / re-evaluation for each render?

Reply:
Updated 👍🏾